docs/data/data-grid/server-side-data/tree-data.md

```diff
@@ -2,14 +2,36 @@
 title: React Server-side tree data
 ---
 
-# Data Grid - Server-side tree data [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')🚧
+# Data Grid - Server-side tree data [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
 
 <p class="description">Tree data lazy-loading with server-side data source.</p>
 
-:::warning
-This feature isn't implemented yet. It's coming.
+To use the server-side tree data, pass the `unstable_dataSource` prop as explained in the [overview section](/x/react-data-grid/server-side-data/), in addition to that passing of some additional props is required for the server-side tree data to work properly.
 
-👍 Upvote [issue #3377](https://github.com/mui/mui-x/issues/3377) if you want to see it land faster.
+- `getGroupKey(row: GridRowModel): string`
 
-Don't hesitate to leave a comment on the same issue to influence what gets built. Especially if you already have a use case for this component, or if you are facing a pain point with the [currently proposed workaround](https://mui.com/x/react-data-grid/tree-data/#children-lazy-loading).
+  Used by the grid to group rows by their parent group. Replaces `getTreeDataPath` used in client-side tree-data.
+  For example, consider this tree structure for tree data.
+
+  ```js
+  - (1) Sarah // groupKey 'Sarah'
+    - (2) Thomas // groupKey 'Thomas'
+  ```
+
+  When `(2) Thomas` is expanded, the `getRows` function will be called with group keys `['Sarah', 'Thomas']`.
+
+- `hasChildren(row: GridRowModel): boolean`
+
+  Used by the grid to check if a row has children on server
+
+- `getChildrenCount?: (row: GridRowModel) => number`
+
+  Used by the grid to determine the number of children of a row on server
+
+Following is a demo of the server-side tree data with the data source which supports filtering, sorting, and pagination on the server. It also supports the caching using the `unstable_dataSourceCache` prop based on the `QueryClient` exposed by `@tanstack/query-core`.
+
+{{"demo": "ServerSideTreeData.js", "bg": "inline"}}
+
+:::info
+The demo above uses a utility `useDemoDataSource` which uses a data generator service to generate data for testing of the application. It exposes a function called `getRows` which could be used directly as `GridDataSource.getRows`.
```

Comment:
Alternative to:
> I'm wondering if we should have a specific section for caching to perhaps better capture some search queries on our docs.

 If we use the demo data in a separate file with the multi-tab support, maybe the demo gets clear enough so we can use this info section to highlight the caching mechanism.

Reply:
The most important is to show the data source implementation on the example.
We need to show the implementation of :
`getGroupKey`, `hasChildren`, `getChildrenCount`, and etc.

Reply:
On an parallel thread:

Alternative to:
> I'm wondering if we should have a specific section for caching to perhaps better capture some search queries on our docs.

 If we use the demo data in a separate file with the multi-tab support, maybe the demo gets clear enough so we can use this info section to highlight the caching mechanism.

Reply:
> I think it could be a nice addition to wrap the demoData and getRows method in a function that resembles more like server side call. That would help users better understand how the feature is supposed to integrate in their setup.

I added more details to the demos, let me know how it feels now.